examples/with-electron-typescript/renderer/pages/detail/[id].tsx

```diff
@@ -30,10 +30,17 @@ const InitialPropsDetail = ({ item, errors }: Props) => {
   )
 }
 
-export const getServerSideProps: GetServerSideProps = async (context) => {
-  const {
-    query: { id },
-  } = context
+export const getStaticPaths: GetStaticPaths = async () => {
+  const items: User[] = await findAll()
+  const paths = items.map((item) => `/detail/${item.id}`)
+  return { paths, fallback: false }
+}
+
+export const getStaticProps: GetStaticProps = async (context) => {
+  const id = context.params?.id
+  if (!id) {
+    throw new Error('Invalid page id')
+  }
```

Comment:
No need to validate the id here.

```suggestion
export const getStaticProps: GetStaticProps = async ({ params }) => {
  const { id } = params
```